src/v/raft/raftgen.json

```diff
@@ -20,6 +21,11 @@
             "input_type": "heartbeat_request",
             "output_type": "heartbeat_reply"
         },
+        {
+            "name": "heartbeat_v2",
```

Comment:
ah that's great, so a node can use either heartbeat or heartbeat_v2 in full compatible mode at all times. 

love this. can you add a test for this explicitly pls. VERY COOL.  cc: @travisdowns on perf improvement here.